To: vim_dev@googlegroups.com
Subject: Patch 7.3.312
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.312 (after 7.3.306)
Problem:    Can't compile with tiny features.
Solution:   Add #ifdef around win_valid().
Files:	    src/buffer.c


*** ../vim-7.3.311/src/buffer.c	2011-09-14 14:43:21.000000000 +0200
--- src/buffer.c	2011-09-14 17:46:24.000000000 +0200
***************
*** 416,422 ****
  #endif
  
      buf_freeall(buf, (del_buf ? BFA_DEL : 0) + (wipe_buf ? BFA_WIPE : 0));
!     if (win_valid(win) && win->w_buffer == buf)
  	win->w_buffer = NULL;  /* make sure we don't use the buffer now */
  
  #ifdef FEAT_AUTOCMD
--- 416,426 ----
  #endif
  
      buf_freeall(buf, (del_buf ? BFA_DEL : 0) + (wipe_buf ? BFA_WIPE : 0));
!     if (
! #ifdef FEAT_WINDOWS
! 	win_valid(win) &&
! #endif
! 			  win->w_buffer == buf)
  	win->w_buffer = NULL;  /* make sure we don't use the buffer now */
  
  #ifdef FEAT_AUTOCMD
*** ../vim-7.3.311/src/version.c	2011-09-14 16:52:02.000000000 +0200
--- src/version.c	2011-09-14 17:27:24.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     312,
  /**/

-- 
You have heard the saying that if you put a thousand monkeys in a room with a
thousand typewriters and waited long enough, eventually you would have a room
full of dead monkeys.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
